gammapy/catalog/fermi.py

```diff
@@ -642,6 +645,53 @@ def flux_points(self):
         table['dnde'] = (e2dnde * e_ref ** -2).to('cm-2 s-1 TeV-1')
         return FluxPoints(table)
 
+    def spatial_model(self, emin=1 * u.TeV, emax=10 * u.TeV):
+        """
+        Source spatial model.
+        """
+        d = self.data
+        flux = self.spectral_model.integral(emin, emax)
+        amplitude = flux.to('cm-2 s-1').value
+
+        pars = {}
+        glon = Angle(d['GLON']).wrap_at('180d')
+        glat = Angle(d['GLAT']).wrap_at('180d')
+
+        if self.pointlike:
+            pars['amplitude'] = amplitude
+            pars['x_0'] = glon.value
+            pars['y_0'] = glat.value
+            return Delta2D(**pars)
+        else:
+            de = self.data_extended
+            morph_type = de['Spatial_Function'].strip()
+
+            if morph_type == 'RadialDisk':
+                pars['x_0'] = glon.value
+                pars['y_0'] = glat.value
+                pars['R_0'] = de['Model_SemiMajor'].to('deg').value
+                pars['amplitude'] =  amplitude / (np.pi * pars['R_0'] ** 2)
+                return Disk2D(**pars)
+            elif morph_type == 'SpatialMap':
+                filename = de['Spatial_Filename'].strip()
+                base = '$GAMMAPY_EXTRA/datasets/catalogs/fermi/Extended_archive_v17/Templates/'
+                template = Template2D.read(base + filename)
+                template.amplitude = amplitude
+                return template
+            elif morph_type == 'RadialGauss':
+                pars['x_mean'] = glon.value
+                pars['y_mean'] = glat.value
+                pars['x_stddev'] = de['Model_SemiMajor'].to('deg').value
+                pars['y_stddev'] = de['Model_SemiMajor'].to('deg').value
+                pars['amplitude'] = amplitude * 1 / (2 * np.pi * pars['x_stddev'] ** 2)
```

Comment:
Such amplitude to integral conversions belong on the model classes, to avoid duplicating the formulae in our codebase, no? Maybe as a static helper function to compute before `__init__` or as an option in `init` or a method to call after init?

In this case just taking `x_stddev` and not considering `y_stddev` is a bug for asymm cases, no?